src/core/ack_tracker.c

```diff
@@ -96,6 +96,64 @@ QuicAckTrackerAddPacketNumber(
         !RangeUpdated;
 }
 
+//
+// This function implements the logic defined in Section 6.2 [draft-ietf-quic-frequence-10]
+// to determine if the reordering threshold has been hit.
+//
+BOOLEAN
+QuicAckTrackerDidHitReorderingThreshold(
+    _In_ QUIC_ACK_TRACKER* Tracker,
+    _In_ uint8_t ReorderingThreshold
+    )
+{
+    if (ReorderingThreshold == 0) {
+        return FALSE;
+    }
+
+    const uint64_t LargestUnacked = QuicRangeGetMax(&Tracker->PacketNumbersToAck);
+    uint64_t LargestReported;  // The largest packet number that could be declared lost 
+
+    //
+    // Largest Reported is equal to the largest packet number acknowledged minus the Reordering Threshold.
+    // If the difference between the largest packet number acknowledged and the Reordering Threshold is smaller than the
+    // smallest packet in the ack tracker, then the largest reported is the smallest packet in the ack tracker.
+    //
+
+    if (Tracker->LargestPacketNumberAcknowledged >= QuicRangeGet(&Tracker->PacketNumbersToAck, 0)->Low + ReorderingThreshold) {
+        LargestReported = Tracker->LargestPacketNumberAcknowledged - ReorderingThreshold + 1;
+    } else {
+        LargestReported = QuicRangeGet(&Tracker->PacketNumbersToAck, 0)->Low;
+    }
+
+    //
+    // Loop through all previous ACK ranges (before last) to find the smallest missing
+    // packet number that is after the largest reported packet number. If the difference
+    // between that missing number and the largest unack'ed number is more than the
+    // reordering threshold, then the condition has been met to send an immediate
+    // acknowledgement.
+    //
+
+    for (uint32_t Index = QuicRangeSize(&Tracker->PacketNumbersToAck) - 1; Index > 0; --Index) {
+        uint64_t SmallestMissing = QuicRangeGetHigh(QuicRangeGet(&Tracker->PacketNumbersToAck, Index - 1)) + 1; // Smallest missing in the previous gap
+        uint64_t RangeStart = QuicRangeGet(&Tracker->PacketNumbersToAck, Index)->Low; // Lowest Packet number in the subrange
+          
+        // 
+        // Check if largest reported packet is missing. In that case, the smalles missing packet becomes the largest reported packet.
```

Comment:
```suggestion
        // Check if largest reported packet is missing. In that case, the smallest missing packet becomes the largest reported packet.
```
Also, please limit line to 80 characters.